### ISSUE TITLE
Expo CLI is using Netinfo version ~2.0.10

### ISSUE THREAD

**Issue by travislang** [expo/expo-cli#983]
# READ THIS FIRST, PLEASE!

We're looking to keep questions on our forums and bug reports on the GitHub repo -- please choose whichever you feel is more appropriate for your issue:

- Post in our community forums (log in with your Expo developer account): https://forums.expo.io/c/help
- File a bug on the Expo CLI repo (for bugs related to the Expo CLI tools): https://github.com/expo/expo-cli/issues/new
- File a bug on the main Expo repo (for other bugs): https://github.com/expo/expo/issues/new
- Request a feature on Canny: https://expo.canny.io/feature-requests

Thanks for reporting the issue you found with Expo CLI!

# Required format for issues

Please make our job easier by filling this template out to completion.

## Description

running `expo install @react-native-community/netinfo` installs version ~2.0.10 which doesn't work with the current docs both on expos website and in the netinfo README.

## Expected Behavior

Being able to use netInfos API like this:
```
const unsubscribe = NetInfo.addEventListener(state => {
  console.log('Connection type', state.type);
  console.log('Is connected?', state.isConnected);
});
```

## Observed Behavior

running the code gives a `Trying to subscribe to unknown event message` warning.
Here is a related issue on netinfo's repo:
https://github.com/react-native-community/react-native-netinfo/issues/93

I'm not sure if this is a mixup between the docs and the installed version but the current API documentation doesn't work with the version of netinfo that gets installed.  The fix is the use the old API:

https://github.com/react-native-community/react-native-netinfo/commit/47faad2470024339ee01b7af68e480eca15eee39?short_path=04c6e90#diff-04c6e90faac2675aa89e2176d2eec7d8

## Environment

Expo CLI 3.0.10 environment info:
    System:
      OS: macOS 10.14.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.9.1 - /usr/local/bin/node
      npm: 6.10.3 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.4 AI-183.5429.30.34.5452501
      Xcode: 10.3/10G8 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^33.0.0 => 33.0.7 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.12.1 => 3.12.1 
    npmGlobalPackages:
      expo-cli: 3.0.10

Also specify:

-iOS simulator - iPhone X

**Comment by fson** [comment]
Good point, looks like the SDK 33 docs at https://docs.expo.io/versions/v33.0.0/sdk/netinfo/ indeed show docs for a newer version of `NetInfo` than what's bundled in SDK 33. (SDK 34 and newer have an updated version of `NetInfo`.)

We'll need to update the SDK 33 docs to reflect the correct API version, which is documented here: https://github.com/react-native-community/react-native-netinfo/tree/v2.0.10#usage
(For example, in the older version it's `NetInfo.addEventListener('connectionChange', listener)` instead of `NetInfo.addEventListener(listener)`).
